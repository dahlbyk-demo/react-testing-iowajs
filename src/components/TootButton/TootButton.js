import React from 'react';

export default class TootButton extends React.Component {
  constructor(){
    super();

    this.state = {
      clicks: 0,
      person: {
        firstName: 'Keith'
      }
    };
  }

  buttonClick = e => {
    this.setState(state => ({ clicks: state.clicks+1}));
    this.props.onClick(e);
  }

  render(){
  const { onClick } = this.props;

  return (
    <button data-testid="toot-button"
      className="btn btn-primary"
       onClick={this.buttonClick}>Toot</button>
  );
  }
}
