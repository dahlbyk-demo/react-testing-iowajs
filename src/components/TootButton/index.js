import TootButton from './TootButton';

export default TootButton;
