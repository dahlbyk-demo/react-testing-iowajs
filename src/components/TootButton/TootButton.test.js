import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import TootButton from './TootButton';

describe('<TootButton />', () => {
  it('can render a <button>', () => {
    let app = shallow(<TootButton />);

    const button = app.find('button');

    expect(button.text()).toBe('Toot');
    expect(button.hasClass('btn')).toBe(true);
    expect(button.hasClass('btn-primary')).toBe(true);

    expect(app.state('clicks')).toBe(0);
  });

  it('calls onClick prop on click and counts clicks', () => {
    let mockClick = jest.fn();

    let app = shallow(<TootButton onClick={mockClick} />);

    const button = app.find('button');
    button.simulate('click');

    expect(mockClick).toHaveBeenCalledTimes(1);
    expect(app.state('clicks')).toBe(1);

    // Demo
    expect(app.state('person')).toHaveProperty('firstName', 'Keith');
    expect(app.state('person').firstName).toBe('Keith');
  });

  it('renders as expected', () => {
    // Also possible with enzyme? https://github.com/adriantoine/enzyme-to-json

    let obj = renderer.create(<TootButton />).toJSON();
    // console.log(obj);

    expect(obj).toMatchSnapshot();
  });
});
