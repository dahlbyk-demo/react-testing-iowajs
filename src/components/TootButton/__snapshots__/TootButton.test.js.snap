// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TootButton /> renders as expected 1`] = `
<button
  className="btn btn-primary"
  data-testid="toot-button"
  onClick={[Function]}
>
  Toot
</button>
`;
